tests/test_spstats.py

```diff
@@ -171,3 +194,30 @@ def test_patches_method(self):
             gsd=diff.res[0],
             area_size=10000
         )
+
+class TestBinning:
+
+    def test_nd_binning(self):
+
+        ref, diff, mask = load_ref_and_diff()
+
+        slp, asp = xdem.coreg.calculate_slope_and_aspect(ref.data.squeeze())
+
+        # 1d binning
+        df = xdem.spstats.nd_binning_scipy(dh=diff.data.flatten(),list_var=[slp.flatten()],list_var_names=['slope'])
+
+        # nmad goes up quite a bit with slope, we can expect a 10 m measurement error difference
+        assert df.nmad.values[-1] - df.nmad.values[0] > 10
+
+        # try custom stat
+        def percentile_80(a):
+            return np.nanpercentile(a, 80)
+
+        xdem.spstats.nd_binning_scipy(dh=diff.data.flatten(),list_var=[slp.flatten()],list_var_names=['slope'], statistics=['count',percentile_80])
```

Comment:
Here and elsewhere, you should have some assert statements.

Reply:
You can see e.g. how it is done with a relatively similar function in volume.py: https://github.com/GlacioHack/xdem/blob/2a389a09ae4d68f7e765e576b1c9afd99021dc92/tests/test_volume.py#L129